arangod/Agency/Agent.cpp

```diff
@@ -327,6 +328,29 @@ index_t Agent::confirmed(std::string const& agentId) const {
 void Agent::reportIn(std::string const& peerId, index_t index, size_t toLog) {
   auto startTime = steady_clock::now();
 
+  if (index == 0) {
+    // This is only the empty case (=heartbeat)
+    MUTEX_LOCKER(locker, _emptyAppendLock);
+    auto n = steady_clock::now();
+    auto lastTime = _lastEmptyAcked[peerId];
```

Comment:
This could create a new entry for `peerId` if it does not yet exist in the map. 
This is probably intentional, but it is worth a code comment.
In addition, in case the map value is updated later, we currently perform an extra lookup in the map.
Suggestion: do the initial lookup and assign the result variable by reference, so later the update does not have to perform an extra lookup.